assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,253 @@
+			'roles must be an array of strings.'
+		);
+		return {
+			payload: {
+				moduleSlug,
+				roles,
+			},
+			type: SET_SHARED_ROLES,
+		};
+	},
+
+	/**
+	 * Saves sharingSettings for dashboard sharing.
+	 *
+	 * Save sharingSettings for dashboard sharing.
+	 * Update ownerID from the response for the modules in the sharingSettings state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `{response, error}`.
+	 */
+	*saveSharingSettings() {
+		const registry = yield Data.commonActions.getRegistry();
+		// TODO: Refactor to use `getSharingSettings` selector
+		// to obtain the `sharingSettings` from the state in 4795.
+		const { sharingSettings } = registry.stores[
+			CORE_MODULES
+		].store.getState();
+
+		const {
+			response,
+			error,
+		} = yield fetchSaveSharingSettingsStore.actions.fetchSaveSharingSettings(
+			sharingSettings
+		);
+
+		// Update module owner IDs in the sharing settings modules.
+		if ( response?.newOwnerIDs ) {
+			for ( const [ slug, ownerID ] of Object.entries(
+				response.newOwnerIDs
+			) ) {
+				const storeName = registry
+					.select( CORE_MODULES )
+					.getModuleStoreName( slug );
+
+				yield registry.dispatch( storeName ).setOwnerID( ownerID );
+			}
+		}
+
+		return { response, error };
+	},
+
+	/**
+	 * Receives sharingSettings for dashboard sharing.
+	 * Stores sharingSettings in the datastore.
+	 *
+	 * Because this is frequently-accessed data, this is usually sourced
+	 * from a global variable (`_googlesitekitDashboardSharingData`), set by PHP
+	 * in the `before_print` callback for `googlesitekit-datastore-site`.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} sharingSettings Sharing settings for modules with `management` and `sharedRoles` properties.
+	 * @return {Object} Action for RECEIVE_GET_SHARING_SETTINGS.
+	 */
+	receiveGetSharingSettings( sharingSettings ) {
+		invariant( sharingSettings, 'sharingSettings is required.' );
+		return {
+			payload: { sharingSettings },
+			type: RECEIVE_GET_SHARING_SETTINGS,
+		};
+	},
+};
+
+const baseReducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_SHARING_MANAGEMENT: {
+			const { moduleSlug, management } = payload;
+
+			return {
+				...state,
+				sharingSettings: {
+					...state.sharingSettings,
+					[ moduleSlug ]: {
+						...state.sharingSettings[ moduleSlug ],
+						management,
+					},
+				},
+			};
+		}
+
+		case SET_SHARED_ROLES: {
+			const { moduleSlug, roles } = payload;
+
+			return {
+				...state,
+				sharingSettings: {
+					...state.sharingSettings,
+					[ moduleSlug ]: {
+						...state.sharingSettings[ moduleSlug ],
+						sharedRoles: roles,
+					},
+				},
+			};
+		}
+
+		case RECEIVE_GET_SHARING_SETTINGS: {
+			const { sharingSettings } = payload;
+
+			return {
+				...state,
+				sharingSettings: {
+					...state.sharingSettings,
+					...sharingSettings,
+				},
+				savedSharingSettings: {
+					...state.savedSharingSettings,
+					...sharingSettings,
+				},
```

Comment:
It shouldn't merge `sharingSettings` and `savedSharingSettings`, it should override it:
```suggestion
				sharingSettings,
				savedSharingSettings: sharingSettings,
```